src/main/java/com/google/sps/data/User.java

```diff
@@ -18,6 +29,43 @@ public static boolean createdRecipe(String userId, String recipeId) {
     Boolean userCreatedRecipe =
         user.getBoolean(DBUtils.getNestedPropertyName(CREATED_RECIPES_KEY, recipeId));
     // note that userCreatedRecipe is a Boolean, not a boolean
-    return userCreatedRecipe != null && userCreatedRecipe == true;
+    return userCreatedRecipe != null && userCreatedRecipe;
+  }
+
+  public static boolean isSavedRecipe(String userId, String recipeId) {
+    DocumentReference userRef = DBUtils.user(userId);
+    ApiFuture<DocumentSnapshot> userFuture = userRef.get();
+    DocumentSnapshot user = DBUtils.blockOnFuture(userFuture);
+
+    if (!user.exists()) {
+      return false;
+    }
+    Boolean userSavedRecipe =
+        user.getBoolean(DBUtils.getNestedPropertyName(SAVED_RECIPES_KEY, recipeId));
+    return userSavedRecipe != null && userSavedRecipe;
+  }
+
+  public static ArrayList<String> createdRecipeIds(String userId) {
+    ArrayList<String> allRecipeIds = DBUtils.allRecipeIds();
+    ArrayList<String> createdRecipeIds = new ArrayList<String>();
+
+    for (String recipeID : allRecipeIds) {
+      if (createdRecipe(userId, recipeID)) {
+        createdRecipeIds.add(recipeID);
+      }
+    }
+    return createdRecipeIds;
+  }
+
+  public static ArrayList<String> savedRecipeIds(String userId) {
```

Comment:
This should not be static. Use the existing `User.uid` property instead of taking in a new `userId`.